homeassistant/components/mqtt/vacuum/schema.py

```diff
@@ -15,18 +19,23 @@
 )
 
 
-def services_to_strings(services, service_to_string):
+def services_to_strings(
+    services: VacuumEntityFeature | int,
+    service_to_string: dict[VacuumEntityFeature, str],
+) -> list[str]:
     """Convert SUPPORT_* service bitmask to list of service strings."""
-    strings = []
+    strings: list[str] = []
     for service in service_to_string:
         if service & services:
             strings.append(service_to_string[service])
     return strings
 
 
-def strings_to_services(strings, string_to_service):
+def strings_to_services(
+    strings: list[str], string_to_service: dict[str, VacuumEntityFeature]
+) -> VacuumEntityFeature | int:
```

Comment:
Do you need to return an `int`?
```suggestion
) -> VacuumEntityFeature:
```

Reply:
0 is not a valid `VacuumEntityFeature`.

Reply:
You're right. I had similar issue on #81903